### PR TITLE
[TASK] Add missing directories

### PR DESCRIPTION
**Pull request by linawolf**
Differentiate more between composer and non-Composer dirs

The following follow ups are planned:

- Move folders of legacy chapters in a separate page
- Explain situation of v12 in main
- Explain situation of v11 in branch 11.5
- Clean up use of packages / local_packages
- TODO: Maybe mention composer-installer v4 in TYPO3 11?
- TODO: Information on migration paths

releases: main, 11.5

### CODE REVIEW

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 1]
```diff
@@ -21,18 +21,76 @@ almost always find the directory structure as outlined below.
...
 
 .. contents::
    :local:
 
+.. _directory-config:
+
+:file:`config/`
+---------------
+
+Composer installations only. TYPO3 configuration directory. This directory
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:43]:
```suggestion
Composer-based installations only. TYPO3 configuration directory. This directory
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 2]
```diff
@@ -21,18 +21,76 @@ almost always find the directory structure as outlined below.
...
+contains installation-wide configuration.
+
+.. _directory-config-sites:
+
+:file:`config/sites/`
+~~~~~~~~~~~~~~~~~~~~~
+
+Composer installations only.  The folder :file:`config/sites` contains
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:51]:
```suggestion
Composer-based installations only.  The folder :file:`config/sites` contains
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 3]
```diff
@@ -21,18 +21,76 @@ almost always find the directory structure as outlined below.
...
+subfolders for each :ref:`site configuration <sitehandling>`.
+
+
+.. _directory-local_packages:
+
+:file:`local_packages/`
+-----------------------
+
+Composer installations only.  Each web site which is run on TYPO3 **should**
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:60]:
```suggestion
Composer-based installations only.  Each web site which is run on TYPO3 **should**
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 4]
```diff
@@ -21,18 +21,76 @@ almost always find the directory structure as outlined below.
...
+have a sitepackage, anextension with a special purpose containing all
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:61]:
```suggestion
have a sitepackage, an extension with a special purpose containing all
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 5]
```diff
@@ -21,18 +21,76 @@ almost always find the directory structure as outlined below.
...
+:file:`config/sites/`
+~~~~~~~~~~~~~~~~~~~~~
+
+Composer installations only.  The folder :file:`config/sites` contains
+subfolders for each :ref:`site configuration <sitehandling>`.
+
+
+.. _directory-local_packages:
+
+:file:`local_packages/`
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:57]: When you introduce now a "local_packages" folder, please take care of other places in the TYPO3 docs, where "packages" is used (to be consistent), for example here: https://docs.typo3.org/m/typo3/guide-installation/main/en-us/MigrateToComposer/BestPractices.html#define-a-local-path-repository
I assume, there are other places.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 6]
```diff
@@ -21,18 +21,76 @@ almost always find the directory structure as outlined below.
...
+-----------------------
+
+Composer installations only.  Each web site which is run on TYPO3 **should**
+have a sitepackage, anextension with a special purpose containing all
+templates, styles, images, etc. needed for the theme.
+
+It is usually stored locally and then symlinked into the :ref:`directory-vendor`
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:64]: That is not true for v11 with composer-installers v3 (the majority) - and as this is merged into v11 it should be `typo3conf/ext`.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 7]
```diff
@@ -21,18 +21,76 @@ almost always find the directory structure as outlined below.
...
+folder. Many projects also need custom extensions that can be stored here.
+
+The folder for local packages has to be defined in the project's :file:`composer.json`
+to be used:
+
+..  code-block:: json
+    :caption: composer.json
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:71]:
```suggestion
    :caption: composer.json

```

The snippet is not rendered, probably due to missing empty line.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 8]
```diff
@@ -21,18 +21,76 @@ almost always find the directory structure as outlined below.
...
 
 :file:`public/`
 ---------------
 
-We assume here that your web root points to a folder called :file:`public` in a Composer-based installation or a legacy 
-installation as is commonly done. Otherwise, replace  :file:`public` with the path to your web root.
+We assume here that your web root points to a folder called :file:`public` in
+a Composer-based installation or a legacy installation as is commonly done.
+Otherwise, replace  :file:`public` with the path to your web root. In legacy
+installations by default all files are located in this folder.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:91]: "In legacy installations, all files are located in this folder by default."

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 9]
```diff
@@ -106,6 +163,15 @@ Amongst others, this directory contains the files :file:`LocalConfiguration.php`
 :file:`AdditionalConfiguration.php`. See chapter
 :ref:`Configuration files <configuration-files>` for details.
 
+.. _directory-public-typo3conf-autoload:
+
+:file:`public/typo3conf/autoload/`
+""""""""""""""""""""""""""""""""""
+
+Legacy installations only. Contains :ref:`autoloading <autoload>` information.
+The files are update each time an extension is installed in the
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:172]:
```suggestion
The files are updated each time an extension is installed via the
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 10]
```diff
@@ -114,13 +180,20 @@ Amongst others, this directory contains the files :file:`LocalConfiguration.php`
...
 
 .. _directory-public-typo3conf-l10n:
-.. _directory-var-labels:
 
-:file:`public/typo3conf/l10n/` or :file:`var/labels`
-"""""""""""""""""""""""""""""""""""""""""""""""""""
+:file:`public/typo3conf/l10n/`
+""""""""""""""""""""""""""""""
+
+Legacy installations only. Directory for extension localisations.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:187]:
```suggestion
Legacy installations only. Directory for extension localizations.
```

As we use American English in docs.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 11]
```diff
@@ -114,13 +180,20 @@ Amongst others, this directory contains the files :file:`LocalConfiguration.php`
...
+Contains all downloaded translation files.
+
+.. _directory-public-typo3conf-sites:
+
+:file:`public/typo3conf/sites/`
+"""""""""""""""""""""""""""""""
 
-Directory for extension localisations. Contains all downloaded translation
-files.
+Legacy installations only. Contains contains
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:195]:
```suggestion
Legacy installations only. Contains
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 12]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
-
 .. _directory-public-typo3temp-var:
 
-:file:`public/typo3temp/var/` or :file:`var/labels`
-"""""""""""""""""""""""""""""""""""""""""""""""""""
+:file:`public/typo3temp/var/`
+"""""""""""""""""""""""""""""
+
+Legacy installations only. Directory for temporary files that contains private
+files (e.g. cached Fluid templates) and should not be publicly available.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:220]:
```suggestion
files (e.g. cache and log files) and should not be publicly available.
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 13]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
+See also :ref:`Environment-configuring-paths` for a more detailed description.
+
+.. _directory-var:
+
+:file:`var/`
+------------
 
-Directory for temporary files that contains private files (e.g. cached
-Fluid templates) and should not be publicly available.
+Composer installations only. Directory for temporary files that contains private files (e.g.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:228]:
```suggestion
Composer-based installations only. Directory for temporary files that contains private files (e.g.
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 14]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
+cache and logs files) and should not be publicly available.
 See also :ref:`Environment-configuring-paths` for a more detailed description.
+
+.. _directory-var-cache:
+
+:file:`var/cache/`
+~~~~~~~~~~~~~~~~~~~
+
+Composer installations only. This directory contains internal files needed
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:237]:
```suggestion
Composer-based installations only. This directory contains internal files needed
```
Hmm, not really "needed for the cache". If the files are not there, they are generated. Perhaps better: ".. contains cache files".

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 15]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
+for the cache.
+
+.. _directory-var-labels:
+
+:file:`var/labels/`
+~~~~~~~~~~~~~~~~~~~
+
+Composer installations only.  The directory :file:`var/labels/` is for extension
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:245]:
```suggestion
Composer-based installations only.  The directory :file:`var/labels/` is for extension
```

"... is for localizations. It contains all downloaded translation files." -> maybe better: "... contains all downloaded translations files." Just omit the "is for" part.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 16]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
+localisations. It contains all downloaded translation files.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:246]:
```suggestion
localizations. It contains all downloaded translation files.
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 17]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
+
+This path can be retrieved from the Environment API, see
+:ref:`Environment-labels-path`.
+
+.. _directory-var-log:
+
+:file:`var/log/`
+~~~~~~~~~~~~~~~~~~~
+
+Composer installations only.  This directory contains log files like the
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:256]:
```suggestion
Composer-based installations only.  This directory contains log files like the
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 18]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
+TYPO3 log, the deprecations log and logs generated by extensions.
+
+.. _directory-vendor:
+
+:file:`vendor/`
+---------------
+
+Composer installations only.  In this directory, which lies outside of
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:264]:
```suggestion
Composer-based installations only.  In this directory, which lies outside of
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 19]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
+`typo3/cms-core` and will therefore be installed into the directory
+:file:`vendor/typo3/cms-core`. The extension `news`, package name
+`georgringer/news` will be installed into the folder
+:file:`vendor/georgringer/news`.
+
+Never put or symlink your extensions manually into this directory as they would
+not be found. Local extensions and sitepackages should be kept in a separate
+folder outside the web root, for example
+:ref:`local_packages <directory-local_packages>`.
+Upon installation , Composer creates a symlink from local_packages to
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:281]: As mentioned above, only true for cms-composer-installers v4+. And mainly not TYPO3 v11.5.

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 20]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
+The directory contains folders for each required vendor and inside each
+vendor directory there is a folder with the different project names.
+
+For example the system extension `core` has the complete package name
+`typo3/cms-core` and will therefore be installed into the directory
+:file:`vendor/typo3/cms-core`. The extension `news`, package name
+`georgringer/news` will be installed into the folder
+:file:`vendor/georgringer/news`.
+
+Never put or symlink your extensions manually into this directory as they would
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:277]: Hmm, would rephrase it: "... as it is managed by Composer and any manual changes are getting lost, e.g. on deployment".

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/DirectoryStructure/Index.rst (hunk 5 above)
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst:57]: We already use local_packages in the sitepackage and would take care of this in a follow up

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 21]
```diff
@@ -21,18 +21,76 @@ almost always find the directory structure as outlined below.
...
 
 :file:`public/`
 ---------------
 
-We assume here that your web root points to a folder called :file:`public` in a Composer-based installation or a legacy 
-installation as is commonly done. Otherwise, replace  :file:`public` with the path to your web root.
+We assume here that your web root points to a folder called :file:`public` in
+a Composer-based installation or a legacy installation as is commonly done.
+Otherwise, replace  :file:`public` with the path to your web root. In legacy
+installations by default all files are located in this folder.
```
Comment (linawolf) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:91]:
```suggestion
installations all files are located in this folder by default..
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 22]
```diff
@@ -139,12 +211,73 @@ for temporary files of extensions and TYPO3 components.
...
+
+For example the system extension `core` has the complete package name
+`typo3/cms-core` and will therefore be installed into the directory
+:file:`vendor/typo3/cms-core`. The extension `news`, package name
+`georgringer/news` will be installed into the folder
+:file:`vendor/georgringer/news`.
+
+Never put or symlink your extensions manually into this directory as they would
+not be found. Local extensions and sitepackages should be kept in a separate
+folder outside the web root, for example
```
Comment (linawolf) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:279]:
```suggestion
Never put or symlink your extensions manually into this directory as it is 
managed by Composer and any manual changes are getting lost, 
for example on deployment. Local extensions and sitepackages 
should be kept in a separate folder outside the web root, for example
```

File: Documentation/ApiOverview/DirectoryStructure/Index.rst (hunk 19 above)
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst:283]: will be done in a follow up

File: Documentation/ApiOverview/DirectoryStructure/Index.rst (hunk 6 above)
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst:64]: Will be dealed with in a follow up

File: Documentation/ApiOverview/DirectoryStructure/Index.rst (hunk 8 above)
Comment (linawolf) [reply Documentation/ApiOverview/DirectoryStructure/Index.rst]: will be solved in a follow up

File: Documentation/ApiOverview/DirectoryStructure/Index.rst [hunk 23]
```diff
@@ -21,18 +21,77 @@ almost always find the directory structure as outlined below.
 Also see :ref:`Environment` for further information, especially how to retrieve
 the paths within PHP code.
 
+Files on project level
+======================
+
+On the top most level, the project level, you can find the files
+:file:`composer.json` which contains requirements for the TYPO3 installation
+and the :file:`composer.lock` which contains information about the concretely
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DirectoryStructure/Index.rst:29]:
```suggestion
and the :file:`composer.lock` which contains information about the concrete
```

**Review by brotkrueml — APPROVED** [review]
